### ISSUE TITLE
Some policies works with the REST API but doesn't work with the Console

### ISSUE THREAD

**Issue by kinjelom** [minio/console#1315]
For example, a user with a policy:
```json
{
    "Version": "2012-10-17",
    "Statement": [{
            "Effect": "Allow",
            "Action": ["s3:*"],
            "Resource": ["arn:aws:s3:::testbucket-*"]
    }]
}
```
can:
- list a bucket `testbucket-abc` using the the REST API and the Console
- list the bucket's objects using the REST API

but can't:
- list the bucket's objects using the Console

Everything works great (using the Console and the REST API) when we allow the user to do all `s3:*` actions on all resources `arn:aws:s3:::*`.

**Comment by grossvater** [comment]
The same behavior, but with specific action: `ListBucket`:

```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "s3:ListBucket"
            ],
            "Resource": [
                "arn:aws:s3:::test"
            ]
        }
    ]
}
```

I can list with rest-api (via client golang sdk), but can't in console. Changing the resource to `*` makes the console list the content of the bucket.

**Comment by Alevsk** [comment]
> ```json
> {
>     "Version": "2012-10-17",
>     "Statement": [{
>             "Effect": "Allow",
>             "Action": ["s3:*"],
>             "Resource": ["arn:aws:s3:::testbucket-*"]
>     }]
> }
> ```

Can you share the mc commands you are using to list buckets & objects? im trying to replicate this using the policy you provided but its not working for me, this is how i tried to reproduce

1. created a new user: testbucket/testbucket123
2. assigned the provided policy to that user
3. ran mc commands:

```
mc alias set testbucket http://localhost:9000 testbucket testbucket123
Added `testbucket` successfully.

mc ls testbucket
mc: <ERROR> Unable to list folder. Access Denied.

mc ls testbucket/testbucket-abc
mc: <ERROR> Unable to list folder. Access Denied.
```

**Comment by kinjelom** [comment]
You have to create a bucket with the `testbucket-` prefix in this case, e.g.: `testbucket-abc`
For example user `test.admin` is allowed `s3:*` on `arn:aws:s3:::testbucket-*`.

- It works when we use `mc`:

```bash
mc alias set testadmin http://localhost:9000 test.admin <test-admin-password>

mc ls -r testadmin/testbucket
mc: <ERROR> Unable to list folder. Access Denied.

mc ls -r testadmin/testbucketabc
mc: <ERROR> Unable to list folder. Access Denied.

mc ls -r testadmin/testbucket-abc
[2021-12-16 09:14:21 UTC] 167KiB small-image.jpg
```

- but it doesn't work when we use the console (web):
![mc-console-bug](https://user-images.githubusercontent.com/2461648/146345151-ebcefb2e-a6d0-4d47-9292-9aae4a67572d.jpg)

**Comment by bpedersen2** [comment]
Maybe related to https://github.com/minio/minio/issues/14099

**Comment by acheglakov** [comment]
Hello!

I also have this issue with console and readonly bucket. I made the same policy as in https://github.com/minio/minio/issues/14099. It works fine, but when I remove "s3:PutObject" I see just a blank screen in the browser. 
mcli works well. 
Minio version 2022-01-08.

**Comment by dvaldivia** [comment]
this should be fixed in #1589

**Comment by xingchenxuup** [comment]
> 

Hello, have you resolved this problem
